pkg/machine/sockets.go

```
@@ -0,0 +1,86 @@
+package machine
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"net"
+	"path/filepath"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// SetSocket creates a new machine file for the socket and assigns it to
+// `socketLoc`
+func SetSocket(socketLoc *VMFile, path string, symlink *string) error {
+	socket, err := NewMachineFile(path, symlink)
+	if err != nil {
+		return err
+	}
+	*socketLoc = *socket
+	return nil
+}
+
+// GetReadySocketPath returns the filepath for the ready socket
+func GetReadySocketPath(runtimeDir, machineName string) string {
+	return filepath.Join(runtimeDir, fmt.Sprintf("%s_ready.sock", machineName))
+}
+
+// ListenAndWaitOnSocket waits for a new connection to the listener and sends
+// any error back through the channel
+func ListenAndWaitOnSocket(errChan *chan error, connChan *chan net.Conn, listener net.Listener) {
```

This should return an error.